ext/SerializationExt.jl

```diff
@@ -0,0 +1,74 @@
+module SerializationExt
+using LRUCache
+using Serialization
+
+function Serialization.serialize(s::AbstractSerializer, lru::LRU{K, V}) where {K, V}
+    Serialization.writetag(s.io, Serialization.OBJECT_TAG)
+    serialize(s, typeof(lru))
+    @assert lru.currentsize == length(lru)
+    serialize(s, lru.currentsize)
+    serialize(s, lru.maxsize)
+    serialize(s, lru.hits)
+    serialize(s, lru.misses)
+    serialize(s, lru.lock)
+    serialize(s, lru.by)
+    serialize(s, lru.finalizer)
+    for (k, val) in lru
+        serialize(s, k)
+        serialize(s, val)
+    end
+end
+
+function Serialization.deserialize(s::AbstractSerializer, ::Type{LRU{K, V}}) where {K, V}
+    currentsize = Serialization.deserialize(s)
+    maxsize = Serialization.deserialize(s)
+    hits = Serialization.deserialize(s)
+    misses = Serialization.deserialize(s)
+    lock = Serialization.deserialize(s)
+    by = Serialization.deserialize(s)
+    finalizer = Serialization.deserialize(s)
+
+    dict = Dict{K, Tuple{V, LRUCache.LinkedNode{K}, Int}}()
+    sizehint!(dict, currentsize)
+    # Create node chain
+    first = nothing
+    node = nothing
+    for i in 1:currentsize
+        prev = node
+        k = deserialize(s)
+        node = LRUCache.LinkedNode{K}(k)
+        val = deserialize(s)
+        sz = by(val)::Int
```

Comment:
I think this is not entirely safe to just recompute the size. In principle, as the `LRU`s are mutable, it is allowed to change the `by` field halfway through filling up the cache. This would then no longer correctly infer the sizes. It's an easy fix though, you just need to (de)serialize (key, val, sz) together.

Reply:
I did this because the size isn't returned when we iterate through the LRU, and it was suggested to iterate through the LRU for serialization. You are right though, good point - I'll work on another method to iterate through the LRU